invokeai/app/services/session_processor/session_processor_default.py

```diff
@@ -16,15 +16,199 @@
 from invokeai.app.util.profiler import Profiler
 
 from ..invoker import Invoker
-from .session_processor_base import SessionProcessorBase
+from .session_processor_base import InvocationServices, SessionProcessorBase, SessionRunnerBase
 from .session_processor_common import SessionProcessorStatus
 
 
+class DefaultSessionRunner(SessionRunnerBase):
+    """Processes a single session's invocations"""
+
+    def __init__(
+        self,
+        on_before_run_node: Union[Callable[[BaseInvocation, SessionQueueItem], bool], None] = None,
+        on_after_run_node: Union[Callable[[BaseInvocation, SessionQueueItem], bool], None] = None,
+    ):
+        self.on_before_run_node = on_before_run_node
+        self.on_after_run_node = on_after_run_node
+
+    def start(self, services: InvocationServices, cancel_event: ThreadEvent):
+        """Start the session runner"""
+        self.services = services
+        self.cancel_event = cancel_event
+
+    def next_invocation(
+        self, previous_invocation: Optional[BaseInvocation], queue_item: SessionQueueItem, cancel_event: ThreadEvent
+    ) -> Optional[BaseInvocation]:
+        invocation = None
+        if not (queue_item.session.is_complete() or cancel_event.is_set()):
+            try:
+                invocation = queue_item.session.next()
+            except Exception as exc:
+                self.services.logger.error("ERROR: %s" % exc, exc_info=True)
+
+                node_error = str(exc)
+
+                # Save error
+                if previous_invocation is not None:
+                    queue_item.session.set_node_error(previous_invocation.id, node_error)
+
+                # Send error event
+                self.services.events.emit_invocation_error(
+                    queue_batch_id=queue_item.batch_id,
+                    queue_item_id=queue_item.item_id,
+                    queue_id=queue_item.queue_id,
+                    graph_execution_state_id=queue_item.session.id,
+                    node=previous_invocation.model_dump() if previous_invocation else {},
+                    source_node_id=queue_item.session.prepared_source_mapping[previous_invocation.id]
+                    if previous_invocation
+                    else "",
+                    error_type=exc.__class__.__name__,
+                    error=node_error,
+                )
```

Comment:
This isn't right, an error when calling `next()` is a session error, not an invocation error.